app/src/main/java/com/hover/stax/balances/BalancesFragment.java

```diff
@@ -105,12 +105,17 @@ private void updateServices(List<Channel> channels, View view) {
 	}
 
 	private void refreshBalances(View v) {
-		if (channelDropdown.getHighlighted() != null) {
+		if (channelDropdown.getHighlighted() != null
```

Comment:
This is not correct. The `setAllRunning` function already causes the view model to fire once the actions have updated if there are none, so setting an observer here is redundant and seriously confuses the already complicated logic in this function.

This should work fine as is. If you don't think it is working please write a failing test first so that we can verify there is a problem.

Reply:
Very costly oversight. Apologies. It actually works without the observer, forgot to remove it.